### ISSUE TITLE
Support non-Microsoft Enterprise SSO Extensions

### ISSUE THREAD

**Issue by AdamUCF** [AzureAD/microsoft-authentication-library-for-objc#1407]
We have an internally developed SSO Extension for iOS that automates sign-on to our on-prem Federation system. Our Azure tenant is federated with this on-prem system. When users attempt to sign-on to a Microsoft app such as Teams or a custom app that uses MSAL, instead of the app triggering our SSO extension it pops the browser and forces the user to log in manually. Is there some way to globally disable the Microsoft SSO Extension support so that it can make use of our corporate extension instead? Even with MS Authenticator uninstalled from the device it refuses to use our SSO Extension. The devices are MDM managed by an on-prem solution if that makes any difference.

**Comment by mattvlasach** [comment]
I'm petty sure this is because the Microsoft Azure AD domain is locked to the Authenticator app bundle IDs using app associated domains.  In other words, per security design by Apple, only allowed bundle IDs may use the SSO extension (the SSO extension reaches out to a well known endpoint to capture that list before it is allowed to function).  

And since everyone uses login.microsoftonline.com for auth, there is no way to define app IDs these per-tenant in the current design of Azure AD even if Microsoft wanted to allow that.  I'm sure custom domains or the like could technically be used, but that would be a lot of effort for Microsoft that I think would work against their strategic direction anyway.

So bottom line: I wouldn't hold your breath on this one.  Your only real option is to create your own IdP that federates with Azure etc., and that isn't a very good option!

**Comment by AdamUCF** [comment]
Yeah the domains I was referring to was our own, not the microsoft domains. 

Buuttt after doing some more digging I figured out that the issue was on our side and it does indeed work.

**Comment by mattvlasach** [comment]
Oh I see, you are able to invoke the SSOE _after_ the federated redirect from Azure AD.  That makes sense!
